session-manager-android/src/main/java/com/web3auth/session_manager_android/types/AES256CBC.java

```diff
@@ -1,37 +1,35 @@
 package com.web3auth.session_manager_android.types;
 
-import com.web3auth.session_manager_android.keystore.KeyStoreManager;
+import org.bouncycastle.util.encoders.Hex;
 
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.math.BigInteger;
+import java.security.InvalidAlgorithmParameterException;
 import java.security.InvalidKeyException;
-import java.security.Key;
 import java.security.NoSuchAlgorithmException;
-import java.security.spec.AlgorithmParameterSpec;
 import java.security.spec.ECFieldFp;
 import java.security.spec.EllipticCurve;
 import java.util.Arrays;
 
+import javax.crypto.BadPaddingException;
 import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
 import javax.crypto.Mac;
+import javax.crypto.NoSuchPaddingException;
 import javax.crypto.spec.IvParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
 
 public class AES256CBC {
     private static final String TRANSFORMATION = "AES/CBC/PKCS5Padding";
-    private final byte[] AES_ENCRYPTION_KEY;
-    private final byte[] ENCRYPTION_IV;
-    private final byte[] MAC_KEY;
-    private final byte[] ENCRYPTION_EPHEM_KEY;
 
-    public AES256CBC(String privateKeyHex, String ephemPublicKeyHex, String encryptionIvHex) throws NoSuchAlgorithmException {
+    public byte[] getAESKey(String privateKeyHex, String ephemPublicKeyHex) throws NoSuchAlgorithmException {
         byte[] hash = SHA512.digest(toByteArray(ecdh(privateKeyHex, ephemPublicKeyHex)));
-        byte[] encKeyBytes = Arrays.copyOfRange(hash, 0, 32);
-        AES_ENCRYPTION_KEY = encKeyBytes;
-        MAC_KEY = Arrays.copyOfRange(hash, 32, hash.length);
-        ENCRYPTION_IV = toByteArray(encryptionIvHex);
-        ENCRYPTION_EPHEM_KEY = toByteArray(ephemPublicKeyHex);
+        return Arrays.copyOfRange(hash, 32, hash.length);
```

Comment:
this is incorrect right.
AES Key is 0-32
MAC Key is 32-length

Reply:
Thanks, fixed.

Will add more tests.

Reply:
Added an additional test for this that makes sure that the aes key and mac key are not the same.